### DISCUSSION TITLE
property type not supported Enum

### DISCUSSION THREAD

**Discussion by Saifallak** [livewire/livewire#5885] [7. Bugs (archived)]
getting this error on any model that has enum casting.

ex:
```php
    protected $casts = [
        'type' => CourseTypeEnum::class,
    ];
```

```php
<?php

namespace App\Enums;

use Spatie\Enum\Laravel\Enum;

/**
 * @method static self ONLINE()
 * @method static self OFFLINE()
 */
final class CourseTypeEnum extends Enum
{
}
```

![image](https://github.com/livewire/livewire/assets/6053156/faa21a78-4509-48ed-9005-526ea9cd0cb2)

**Comment by nexxai** [comment]
According to [the docs](https://livewire.laravel.com/docs/properties#supported-property-types), Enums are not a supported property type but you can support them yourself with wireables or Synthesizers.

**Reply by nuernbergerA** [reply]
native PHP `Enum`s are supported.

you are using https://github.com/spatie/laravel-enum/blob/main/src/Enum.php which is a regular class.
maybe spatie is open for a PR to support livewire synths

**Reply by Saifallak** [reply]
Opened issue in spatie https://github.com/spatie/laravel-enum/issues/101

**Reply by nexxai** [reply]
> native PHP `Enum`s are supported.

If that's true, can you update the linked doc page to include all of the native supported types, since the page makes it clear and calls out the fact that "Livewire supports a limited set of property types"?  When a warning like that is made, I think most people would naturally treat any given examples (like the ones specified below that paragraph) as an exhaustive list.

**Reply by joshhanley** [reply]
@nexxai yep, that docs page should be a full list, but it seems native enums were missed by mistake. Please remember that we are still in beta so the package and the docs won't be perfect yet. PRs are welcome and appreciated to improve the docs! 🙂

**Reply by nexxai** [reply]
Oh for sure. I didn't mean for that to come off snarky, and if it did I apologize. Was just explaining my thought process (probably unnecessarily).  

I'm out right now but if it's not done when I get home later tonight, I will submit a PR to add them to the docs. :)

**Reply by joshhanley** [reply]
@nexxai no worries! Ok great thanks, much appreciated 🙂

**Comment by chrisrhymes** [comment]
Just wondered if anyone has a code example of making a native PHP enum wireable in Livewire v3? I am having difficulty making it work. Thank you.

**Reply by nicko170** [reply]
Sure.

```php
<?php

namespace App\Enums;

use Livewire\Wireable;
use App\Traits\WireableEnum;

enum UnitEnum implements Wireable
{
    use WireableEnum;

    case METRIC;
    case IMPERIAL;
}

```

```php
<?php

namespace App\Traits;

use ReflectionEnum;
use ReflectionException;

trait WireableEnum
{
    public function toLivewire(): array
    {
        return [
            'case' => $this->name,
        ];
    }

    /**
     * @throws ReflectionException
     */
    public static function fromLivewire($value): ?static
    {
        return (new ReflectionEnum(static::class))
            ->getCase($value['case'])
            ->getValue();
    }
}
```
